### PR TITLE
feat(Grid): add props interface 

### PR DESCRIPTION
**Pull request by mnajdova**
This PR introduces props interface for the Grid component, as part of #117

### CODE REVIEW

File: src/lib/customPropTypes.tsx
Comment (levithomason) [inline src/lib/customPropTypes.tsx]: Let's not mix TypeScript typings and PropTypes.  They both shared the name "type" but they are entirely different.

Propose we keep typings in `/types`.

File: src/themes/teams/components/Grid/gridStyles.ts
Comment (levithomason) [inline src/themes/teams/components/Grid/gridStyles.ts]: I also think we should hold off on typing styles until we figure out how we're going to do that.  Probably warrants a separate PR, some testing, and discussion.

**Review by levithomason — APPROVED** [review]
Approving with change requests, see comments.

File: types/utils.d.ts
Comment (kuzhelov) [inline types/utils.d.ts:5]: 👍 we should put here `Extendable` as well

Comment (mnajdova) [reply types/utils.d.ts:5]: Will put it in the Button PR, as there I will finally have all components so that I can update them accordiongly. That is why I introduced just this type here.

Comment (kuzhelov) [reply types/utils.d.ts:5]: got it, thanks!

**Review by kuzhelov — APPROVED** [review]

**Pull request merged**
